modules/api/pkg/ee/clusterbackup/storage-location/provider.go

```diff
@@ -200,3 +256,14 @@ func (p *BackupStorageProvider) patchCredentials(ctx context.Context, secretName
 	}
 	return p.privilegedClient.Patch(ctx, updated, ctrlruntimeclient.MergeFrom(existing))
 }
+
+func (p *BackupStorageProvider) getImpersonatedClient(userInfo *provider.UserInfo) (ctrlruntimeclient.Client, error) {
+	if userInfo == nil {
+		return nil, errors.New("a user is missing but required")
+	}
```

Comment:
I think since you are checking for userInfo to not be nil always before calling this func, this can be removed.